src/math/ec_keys.cpp

```diff
@@ -128,7 +162,7 @@ ec_secret create_nonce(ec_secret secret, hash_digest hash)
     {
         V = hmac_sha256_hash(V, K);
 
-        if (verify_private_key(V))
+        if (0 == index--)
```

Comment:
IMHO, `index == 0` and then `index -= 1` would save everyone the hassle of having to remember prefix/postfix operators.